include/jni/functions.hpp

```diff
@@ -589,7 +589,7 @@ namespace jni
     inline UniqueEnv AttachCurrentThread(JavaVM& vm)
        {
         JNIEnv* result;
-        CheckErrorCode(vm.AttachCurrentThread(&result, nullptr));
+        CheckErrorCode(vm.AttachCurrentThread(reinterpret_cast<void**>(&result), nullptr));
```

Comment:
The [Java SE 6 specification](http://docs.oracle.com/javase/6/docs/technotes/guides/jni/spec/invocation.html) did actually specify `JNIEnv **`, but apparently the reference jni.h used `void**`, and some implementation chose to follow the reference jni.h rather than the specification. In Java SE 7, [the specification](http://docs.oracle.com/javase/7/docs/technotes/guides/jni/spec/invocation.html) was changed to `void **`.

https://bugs.openjdk.java.net/browse/JDK-6569899

This puts us in a bind, because using one or the other will fail for implementations whose jni.h make the opposite choice. Therefore we will need to use type traits to determine whether or not to perform a cast or not. Are you up for implementing that?